lib/index.js

```
@@ -293,6 +295,10 @@ exports.Definitions = class {
                 segment = `${segment}; SameSite=${definition.isSameSite}`;
             }
 
+            if (definition.isPartitioned && definition.isSecure && definition.isSameSite === 'None') {
```

Silently ignoring an option is not how hapi usually works. I would much rather see it `throw  Boom.badImplementation()` if it is not a valid combination.

---

Hi @kanongil , thank you for reviewing.
I've applied your suggestion and updated tests.
I wasn't able to run tests though (I'm unable to run tests on master either).

I'll need your help to run tests before merging.

```
statehood feature/cookie-partition % npm run test

> @hapi/statehood@8.1.1 test
> lab -a @hapi/code -t 100 -L

/Users/leandro/Projects/statehood/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2156
                const formattedValue = JSON.stringify(error.data);
                                            ^

TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
```

---

I see that the CI pipeline failed.
Not being able to run my tests locally is blocking me.

What I'm doing:
```sh
node -v
# v20.11.0
npm -v
# 10.2.4
git checkout master
rm -rf node_modules && rm package-lock.json
npm i
npm run test
```
Any thoughts on what I'm doing wrong?